src/test/java/com/fastcampus/projectboard/util/test/FormDataEncoderTest.java

```diff
@@ -0,0 +1,79 @@
+package com.fastcampus.projectboard.util.test;
+
+import com.fastcampus.projectboard.util.FormDataEncoder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+@DisplayName("테스트 도구 - Form 데이터 인코더")
+@Import({FormDataEncoder.class, ObjectMapper.class})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, classes = Void.class)
+class FormDataEncoderTest {
+
+    private final FormDataEncoder formDataEncoder;
+
+    public FormDataEncoderTest(@Autowired FormDataEncoder formDataEncoder) {
+        this.formDataEncoder = formDataEncoder;
+    }
+
+    @DisplayName("객체를 넣으면, url encoding 된 form body data 형식의 문자열을 돌려준다.")
+    @Test
+    void givenObject_whenEncoding_thenReturnsFormEncodedString() {
+        // Given
+        TestObject obj = new TestObject(
+                "This 'is' \"test\" string.",
+                List.of("hello", "my", "friend").toString().replace(" ", ""),
+                String.join(",", "hello", "my", "friend"),
+                null,
+                1234,
+                3.14,
+                false,
+                BigDecimal.TEN,
+                TestEnum.THREE
+        );
+
+        // When
+        String result = formDataEncoder.encode(obj);
+
+        // Then
+        assertThat(result).isEqualTo(
+                """
+                    str=This%20'is'%20%22test%22%20string.
+                    &listStr1=%5Bhello,my,friend%5D
+                    &listStr2=hello,my,friend
+                    &nullStr
+                    &number=1234
+                    &floatingNumber=3.14
+                    &bool=false
+                    &bigDecimal=10
+                    &testEnum=THREE
+                    """.replace("\n", "") // 자동 줄 띄우기 삭제 만약 인코딩 내용에 줄 띄우기가 포함된다면 replace 대신 다른 방법을 사용해야함
+        );
```

Comment:
요렇게 할 수도 있습니다.

```suggestion
        assertThat(result).isEqualTo("""
                str=This%20'is'%20%22test%22%20string.\
                &listStr1=%5Bhello,my,friend%5D\
                &listStr2=hello,my,friend\
                &nullStr\
                &number=1234\
                &floatingNumber=3.14\
                &bool=false\
                &bigDecimal=10\
                &testEnum=THREE""");
```

이 스타일은 들여쓰기에 이득이 있고 `replace()`를 안 해도 됩니다. 텍스트 블록 기호 끝에 `);`와 같이 줄바꿈 없이 끝나는 스타일은 공식 문서 예제를 참고했습니다.
참고로 기존 방식은 데이터 안의 `\n`을 단순히 모두 치환하는데, 이런 방식은 서비스 로직에서 사용할 때 주의해야 합니다.
데이터 중에 사용자가 입력한 `\n`가 포함되어 있을 경우(예: 게시글 본문) 사용자의 데이터가 변조될 수 있습니다.

관련 문서는 공식 문서의 `New Escape Sequences`를 참고하세요.

* https://docs.oracle.com/en/java/javase/21/text-blocks/index.html#new-escape-sequences

Reply:
`replace()`를 사용하게 된다면 데이터의 모든 줄바꿈을 바꾸기때문에 의도된 줄바꿈에 대한 걱정이 있었으나
위 방법처럼하면 좋은 방법이 되겠군요!

`Commit suggestion` 하겠습니다. 감사합니다!